ingestion/src/metadata/ingestion/source/database/postgres_query_parser.py

```diff
@@ -0,0 +1,103 @@
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Postgres usage module
+"""
+import csv
+import traceback
+from abc import ABC
+from datetime import datetime
+from typing import Iterable, Optional
+
+from metadata.generated.schema.entity.services.connections.database.postgresConnection import (
+    PostgresConnection,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.tableQuery import TableQueries, TableQuery
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.source.database.query_parser_source import QueryParserSource
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class PostgresQueryParserSource(QueryParserSource, ABC):
+    """
+    Postgres base for Usage and Lineage
+    """
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        """Create class instance"""
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: PostgresConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, PostgresConnection):
+            raise InvalidSourceException(
+                f"Expected PostgresConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_table_query(self) -> Optional[Iterable[TableQuery]]:
+        """
+        If queryLogFilePath available in config iterate through log file
+        otherwise execute the sql query to fetch TableQuery data
+        """
+        try:
+            if self.config.sourceConfig.config.queryLogFilePath:
+                query_list = []
+                with open(self.config.sourceConfig.config.queryLogFilePath, "r") as fin:
+
+                    for i in csv.DictReader(fin):
+                        query_dict = dict(i)
+
+                        analysis_date = (
+                            datetime.utcnow()
+                            if not query_dict.get("session_start_time")
+                            else datetime.strptime(
+                                query_dict.get("session_start_time"),
+                                "%Y-%m-%d %H:%M:%S+%f",
+                            )
+                        )
+
+                        query_dict["aborted"] = (
+                            True if query_dict["sql_state_code"] == "00000" else False
```

Comment:
I believe this `True if boolean else False` is redundant. We could have `query_dict["aborted"] = query_dict["sql_state_code"] == "00000"`